src/net/http/server.clj

```diff
@@ -146,40 +144,48 @@
       (channelInactive [^ChannelHandlerContext ctx]
         (close-channel ctx (:chan @state) true))
       (channelRead [^ChannelHandlerContext ctx msg]
-        (cond
-          (instance? HttpRequest msg)
-          (let [request (http/->request msg)]
-            (cond 
-              (bad? request)
-              (notify-bad-request! handler msg ctx (:chan @state)
-                                   "Trailing content on request")
-              
-              (= :net.http/unparsable-request request)
-              (notify-bad-request! handler msg ctx (:chan @state)
-                                   "Unparsable request")
-              
-              :else
-              (let [in      (a/chan inbuf)
-                    version (http/protocol-version msg)
-                    bodyreq (assoc request
-                                   :body in
-                                   :is-100-continue-expected? (HttpUtil/is100ContinueExpected msg)
-                                   :send-100-continue! (send-100-continue-fn ctx msg))]
-                (get-response (vswap! state assoc
-                                      :version version
-                                      :chan in
-                                      :request bodyreq)
-                              handler ctx executor))))
-
-          (chunk/content-chunk? msg)
-          (chunk/enqueue (:chan @state) ctx msg)
-
-          :else
-          (do
-            (buf/release msg)
-            (chan/close! ctx)
-            (a/close! (:chan @state))
-            (throw (IllegalArgumentException. "unhandled message chunk on body channel"))))))))
+        (let [chan (:chan @state)
+              close-resources (fn []
+                                (when (buf/buffer? msg)
+                                  (buf/release msg))
+                                (chan/close! ctx)
+                                (when chan
+                                  (a/close! chan)))]
+          (cond
+            (instance? HttpRequest msg)
+            (let [request (http/->request msg)
+                  version (http/protocol-version msg)
+                  respond! (partial write-response ctx version)]
```

Comment:
`write-response` will write response to the Netty `Channel` that represent the connection.
Are we sure this `Channel` cannot be closed ? Otherwise it will throw => Netty will call `exceptionCaught` handler => our services will report Sentry in this scenario.

I am wondering in which code path do we enter when connection is closed on Client side. BadRequest is generated by Netty `HttpObjectDecoder` when not able to parse the request. Maybe `HttpObjectDecoder` can throw when it is not able to read bytes because connection has been closed ?

I think we are better to **try** to send the response (aka wrapping the `write-response` in a `try/catch` bloc)